middlewares/auth.js

```diff
@@ -0,0 +1,19 @@
+const validateToken = async (req, res, next) => {
+  try {
+    const jwtToken = req.headers.authorization;
+    const decoded = jwt.verify(jwtToken, process.env.SECRET_KEY);
+
+    req.user = decoded.payLoad.id;
+    next();
+  } catch (err) {
+    const error = new Error('INVALID_TOKEN');
+    error.statusCode = err.statusCode || 500;
+    next(error);
+  }
+
+  return validateToken;
+};
+
+module.exports = {
+  validateToken,
+};
```

Comment:
다음과 같이 수정해보는 것은 어떨까요?

```suggestion

const loginRequired = async (req, res, next) => {

	// 1) Getting token and check of it's there
  const accessToken = req.headers.authorization

	if (!accessToken) {
		const error = new Error('NEED_ACCESS_TOKEN')
		error.statusCode = 401
		
		return res.status(error.statusCode).json({message: error.message})
	}

  // 2) Verification token
  const decoded = await jwt.verify(accessToken, process.env.JWT_SECRET);

  // 3) Check if user still exists
	const user = await userService.getUserById(decoded.id)

	if (!user) {
		const error = new Error('USER_DOES_NOT_EXIST')
		error.statusCode = 404
		
		return res.status(error.statusCode).json({message: error.message})
	}

  // 4) GRANT ACCESS
  req.user = user;
  next();
}

module.exports = { loginRequired }
```